h2/src/main/org/h2/expression/function/Function.java

```diff
@@ -1417,6 +1417,10 @@ protected Value getValueWithArgs(Session session, Expression[] args) {
         case TRANSLATE: {
             String matching = v1.getString();
             String replacement = v2.getString();
+            if( Mode.ModeEnum.DB2.equals(database.getMode().getEnum())){
+                matching = v2.getString();
+                replacement = v1.getString();
```

Comment:
Use if-else here to avoid double invocation of `Value.getString()` in DB2 mode for each argument or swap `matching` and `replacement` with a temporary variable. `Value.getString()` allocates a new `java.lang.String` if argument is not a `ValueString` (`VARCHAR`). Somebody may use this function with `CLOB` or other data type.